index.ts

```diff
@@ -0,0 +1,42 @@
+module browser_fs {
+    export class SourceFile {
+        fileName: string;
+        text: string = "";
+
+        constructor(_fileName: string) {
+            this.fileName = _fileName;
+        }
+    };
+
+    class ScriptTarget {};
+
+    export class FileSystem {
+        fileSystem: { [fileName: string]: SourceFile } = {};
+
+        getSourceFile(fileName: string, languageVersion: ScriptTarget, 
+                                        onError?: (message: string) => void): SourceFile
+        {
+            return this.fileSystem[fileName];
+        }
+
+        getCanonicalFileName(fileName: string): string {
+            return fileName;
+        }
+
+        writeFile(fileName: string, data: string, writeByteOrderMark: boolean,
+                  onError?: (message: string) => void, sourceFiles?: SourceFile[]): void
```

Comment:
I don't think we need a complicated interface. Let's just start with a simple one